dev/src/reference.ts

```diff
@@ -1875,21 +1947,44 @@ export class Query<T = DocumentData> {
       };
     }
 
-    if (this._queryOptions.fieldOrders.length) {
-      const orderBy: api.StructuredQuery.IOrder[] = [];
-      for (const fieldOrder of this._queryOptions.fieldOrders) {
-        orderBy.push(fieldOrder.toProto());
+    if (this._queryOptions.limitType === LimitType.Last) {
+      if (!this._queryOptions.hasFieldOrders()) {
+        throw new Error(
+          'limitToLast() queries require specifying at least one orderBy() clause.'
+        );
+      }
+
+      structuredQuery.orderBy = this._queryOptions.fieldOrders!.map(order => {
+        // Flip the orderBy directions since we want the last results
+        const dir =
+          order.direction === 'DESCENDING' ? 'ASCENDING' : 'DESCENDING';
+        return new FieldOrder(order.field, dir).toProto();
+      });
+
+      // Swap the cursors to match the now-flipped query ordering.
+      structuredQuery.startAt = this.toCursor(
```

Comment:
I'd suggest keeping the cursor reverse logic here instead of inside `toCursor`, because the order flipping is also done in this block and that makes the intention more explicit IMHO, and that is also how we do it in other SDKs.

Reply:
I am not a huge fan of the code this creates, but it makes sense to keep in aligned with the rest of our SDKs.